### PR TITLE
Fixes #6466 - New ErrorActionPreference Experimental Feature

### LINKED ISSUE #6466: `$ErrorActionPreference` doesn't affect stderr output for native commands
# Summary of the new document or enhancement

The PR https://github.com/PowerShell/PowerShell/pull/13361 made a breaking change that `$ErrorActionPreference` doesn't affect stderr output for native commands.
Also the error records generated from the stderr output of native commands are not put in `$Error` and variable specified by `-ErrorVariable`.

- List of articles that need to be updated:
- New articles that need to be created:
- Link(s) to related code PR(s) in the PowerShell/PowerShell repo: https://github.com/PowerShell/PowerShell/pull/13361
- Link(s) to related issue(s) in the PowerShell/PowerShell repo: https://github.com/PowerShell/PowerShell/issues/3996

### PR DESCRIPTION
**Pull request by chasewilson**
# PR Summary

Adds information about new experimental feature

## PR Context

Fixes #6466 
Fixes [AB#1759889](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1759889)

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [x] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [ ] Version 7.1 preview content
- [ ] Version 7.0 content
- [ ] Version 6 content
- [ ] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

**Comment by chasewilson** [comment]
#sign-off @DCtheGeek

File: reference/docs-conceptual/learn/experimental-features.md [hunk 1]
```diff
@@ -165,13 +166,23 @@ If a PSDrive path that uses the FileSystem provider is passed to a native comman
...
 operating system.
 
 - If the path is not a PSDrive or `~` (on Windows), then path normalization doesn't occur
 - If the path is in single quotes, then it's not resolved and treated as literal
 
+## PSNotApplyErrorActionToStderr
+
+When this experimental feature is enabled, the preference variable `$ErrorActionPreference` does not
+affect the `stderr` output for native commands and does not write the error records from native
+commands are not written to the `$Error` variable.
```
Comment (DCtheGeek) [inline reference/docs-conceptual/learn/experimental-features.md:180]: The second half seems like some extra words got thrown in.  Can you correct or rephrase?

File: reference/docs-conceptual/learn/experimental-features.md [hunk 2]
```diff
@@ -165,13 +166,23 @@ If a PSDrive path that uses the FileSystem provider is passed to a native comman
...
+
+Many native commands write to `stderr` as an alternative stream for additional information. This
```
Comment (DCtheGeek) [inline reference/docs-conceptual/learn/experimental-features.md:182]: Give "This" a noun subject.

Comment (chasewilson) [reply reference/docs-conceptual/learn/experimental-features.md:182]: Fixed and made some other grmmatical changes too

File: reference/docs-conceptual/learn/experimental-features.md (hunk 1 above)
Comment (chasewilson) [reply reference/docs-conceptual/learn/experimental-features.md:180]: Rephrased entir section, let me know what you think

File: reference/docs-conceptual/learn/experimental-features.md (hunk 2 above)
Comment (DCtheGeek) [reply reference/docs-conceptual/learn/experimental-features.md:182]: Not a fan of semi-colons, but I'll allow it. ;)

File: reference/docs-conceptual/learn/experimental-features.md [hunk 3]
```diff
@@ -165,13 +166,23 @@ If a PSDrive path that uses the FileSystem provider is passed to a native comman
...
 operating system.
 
 - If the path is not a PSDrive or `~` (on Windows), then path normalization doesn't occur
 - If the path is in single quotes, then it's not resolved and treated as literal
 
+## PSNotApplyErrorActionToStderr
+
+When this experimental feature is enabled, error records from native commands are not written to the
+`$Error` variable and the preference variable `$ErrorActionPreference` does not affect the `stderr`
+out put for native commands.
```
Comment (DCtheGeek) [inline reference/docs-conceptual/learn/experimental-features.md:180]: "out put" should be one word, right?

Comment (chasewilson) [reply reference/docs-conceptual/learn/experimental-features.md:180]: Dangit! You're right! Curse my fat fingers.

File: reference/docs-conceptual/learn/experimental-features.md [hunk 4]
```diff
@@ -165,13 +166,23 @@ If a PSDrive path that uses the FileSystem provider is passed to a native comman
...
 operating system.
 
 - If the path is not a PSDrive or `~` (on Windows), then path normalization doesn't occur
 - If the path is in single quotes, then it's not resolved and treated as literal
 
+## PSNotApplyErrorActionToStderr
+
+When this experimental feature is enabled, error records from native commands are not written to the
+`$Error` variable and the preference variable `$ErrorActionPreference` does not affect the `stderr`
+output for native commands.
```
Comment (mklement0) [inline reference/docs-conceptual/learn/experimental-features.md:180]: It is important to note that this only applies _when stderr is explicitly redirected with `2>`_ - by default, stderr output does _not_ touch PowerShell's error stream and is passed through to the host, so nothing is recorded in `$Error` and `$ErrorActionPreference` doesn't come into play.


That explicitly sending stderr output _elsewhere_ (e.g., `2>$null` or `2>errs.txt`) suddenly _also_ records it in `$Error` and makes it subject to `$ErrorActionPreference`, as is currently the case, never made any sense, which is why this shouldn't be an experimental feature but a bug fix.

**Pull request merged**